source/qml/Samples/QmlVisualizer/AdaptiveCardQml/MultiLineTextInputRender.qml

```diff
@@ -0,0 +1,84 @@
+﻿import "AdaptiveCardUtils.js" as AdaptiveCardUtils
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.3
+
+Rectangle {
+
+    property bool _showErrorMessage
+    property alias inputtextTextField: _inputtextTextField
+
+    width: parent.width
+    height: CardConstants.inputTextConstants.multiLineTextHeight
+    color:'transparent'
+
+    ScrollView {
+        anchors.fill:parent
+        ScrollBar.vertical.interactive:true
+        ScrollBar.horizontal.interactive:false
+        ScrollBar.horizontal.visible:false
+
+        TextArea {
+            property bool outerShowErrorMessage: _showErrorMessage
+            id: _inputtextTextField
+            wrapMode:Text.Wrap
+            selectByMouse:true
+            selectedTextColor:'white'
+            color: CardConstants.inputFieldConstants.textColor
+            placeholderTextColor: CardConstants.inputFieldConstants.placeHolderColor
+            leftPadding: CardConstants.inputFieldConstants.textHorizontalPadding
+            rightPadding: CardConstants.inputFieldConstants.textHorizontalPadding
+            height:_inputtextTextFieldWrapper.height// need to check for strech height
+            width:_inputtextTextFieldWrapper.width
+            Accessible.role:Accessible.EditableText
+            onTextChanged: { remove(_maxLength, length)
+            if(_isRequired || _regex != "" ) {
+                    validate()
+                }
+                _submitValue = text
+            }
+
+            background:Rectangle{
+                radius: CardConstants.inputFieldConstants.borderRadius
+                id: _multilinetextidTextFieldBackground
+                color: CardConstants.inputFieldConstants.backgroundColorNormal
+                border.color:_inputtextTextField.outerShowErrorMessage? CardConstants.inputFieldConstants.borderColorOnError:_inputtextTextField.activeFocus? CardConstants.inputFieldConstants.borderColorOnFocus : CardConstants.inputFieldConstants.borderColorNormal
+                border.width: CardConstants.inputFieldConstants.borderWidth
+            }
+
+            function colorChange(colorItem,focusItem,isPressed) {
+                if (isPressed && !focusItem.outerShowErrorMessage) {
```

Comment:
Can we can move this function to Rectangle Component? We can remove `outerShowErrorMessage` property if we move this to Rectangle Component